microprofile/connectors/kafka/src/main/java/io/helidon/microprofile/connectors/kafka/KafkaConnectorFactory.java

```diff
@@ -0,0 +1,130 @@
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.connectors.kafka;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.BeforeDestroyed;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+
+import io.helidon.config.Config;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.spi.Connector;
+import org.eclipse.microprofile.reactive.messaging.spi.IncomingConnectorFactory;
+import org.eclipse.microprofile.reactive.messaging.spi.OutgoingConnectorFactory;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.eclipse.microprofile.reactive.streams.operators.SubscriberBuilder;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+/**
+ * Implementation of Connector as described in the MicroProfile Reactive Messaging Specification.
+ */
+@ApplicationScoped
+@Connector(KafkaConnectorFactory.CONNECTOR_NAME)
+class KafkaConnectorFactory implements IncomingConnectorFactory, OutgoingConnectorFactory {
+
+    /**
+     * Microprofile messaging Kafka connector name.
+     */
+    static final String CONNECTOR_NAME = "helidon-kafka";
+    private static final String POOL_SIZE = "kafka.connector.pool.size";
+    private static final Logger LOGGER = Logger.getLogger(KafkaConnectorFactory.class.getName());
+    private final ScheduledExecutorService scheduler;
+    private final Queue<BasicKafkaConsumer<Object, Object>> consumers = new LinkedList<>();
+    private final Queue<BasicKafkaProducer<Object, Object>> producers = new LinkedList<>();
+
+    @Inject
+    KafkaConnectorFactory(Config config) {
+        scheduler = Executors.newScheduledThreadPool(config.get(POOL_SIZE)
+                .asInt().asOptional().orElseGet(() -> 10));
+    }
+
+    /**
+     * Called when container is terminated.
+     *
+     * @param event termination event
+     */
+    public void terminate(@Observes @BeforeDestroyed(ApplicationScoped.class) Object event) {
+        LOGGER.info("Terminating KafkaConnectorFactory...");
+        // Stops the scheduler first to make sure no new task will be triggered meanwhile consumers are closing
+        scheduler.shutdown();
+        BasicKafkaConsumer<Object, Object> consumer;
+        while ((consumer = consumers.poll()) != null) {
+            consumer.close();
+        }
+        BasicKafkaProducer<Object, Object> producer;
+        while ((producer = producers.poll()) != null) {
+            producer.close();
+        }
+        LOGGER.info("KafkaConnectorFactory terminated successfuly");
```

Comment:
Do not use `info` log level so much. If you want to log an info message, just log a single one in this method.